tailscale/client_test.go

```diff
@@ -756,3 +756,52 @@ func TestErrorData(t *testing.T) {
 		assert.Empty(t, tailscale.ErrorData(io.EOF))
 	})
 }
+
+func TestClient_ValidateACL(t *testing.T) {
+	t.Parallel()
+
+	client, server := NewTestHarness(t)
+
+	acl := tailscale.ACL{
+		ACLs: []tailscale.ACLEntry{
+			{
+				Action: "accept",
+				Ports:  []string{"*:*"},
+				Users:  []string{"*"},
+			},
+		},
+		TagOwners: map[string][]string{
+			"tag:example": {"group:example"},
+		},
+		Hosts: map[string]string{
+			"example-host-1": "100.100.100.100",
+			"example-host-2": "100.100.101.100/24",
+		},
+		Groups: map[string][]string{
+			"group:example": {
+				"user1@example.com",
+				"user2@example.com",
+			},
+		},
+		Tests: []tailscale.ACLTest{
+			{
+				User:  "user1@example.com",
+				Allow: []string{"example-host-1:22", "example-host-2:80"},
+				Deny:  []string{"exapmle-host-2:100"},
+			},
+			{
+				User:  "user2@example.com",
+				Allow: []string{"100.60.3.4:22"},
```

Comment:
Tailscale addresses are 100.64.0.0/10, so 100.60.3.4 is not actually a Tailscale address it is an IP which might be reachable by a subnet router.

Recommend choosing a different IP, something within 100.64.0.0/10 if this is intended to be a Tailscale IP or, if it is a public IP, choosing one which doesn't start with `100`